tests/github_client/mod.rs

```diff
@@ -0,0 +1,543 @@
+//! `GithubClient` tests.
```

Comment:
Yea, the JSON files are a mess. I'll look into organizing it.  I think I would prefer to avoid keeping them external, since I think that would make it more difficult to manage.

I'm not sure how feasible it is to make it run online (particularly on CI). Some of the CI complications are authentication and handling concurrent jobs. I also wanted to ensure this is as easy as possible for contributors to run.

Reply:
I've been thinking about this off and on, and I'm feeling like the server tests are going to be a pain to maintain. They aren't too difficult to create, but if someone modifies some behavior in triagebot that perturbs them in any way, they are going to need to figure out how to recreate the tests which I think is just going to be too painful.

I think I'm going to need to figure out a different strategy.

If I dropped the server tests from this PR, and left just the GithubClient and database tests, would you be more willing to merge it? I think those should be much more maintainable, and at least provide some increase in testing.

Reply:
In general -- happy to experiment, the main "blocker" for this PR has (from my perspective) been that you feel happy with it  -- I would like to understand the tests as well, and feel OK updating them, but the main thing is that you seem good since we can always just not run them / delete them.

So reducing scope seems fine.